nullaway/src/test/java/com/uber/nullaway/EnsuresNonNullIfTests.java

```diff
@@ -0,0 +1,801 @@
+            "  }",
+            "}")
+        .addSourceLines(
+            "ChildLevelOne.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "import com.uber.nullaway.annotations.EnsuresNonNullIf;",
+            "class ChildLevelOne extends SuperClass {",
+            "  @Nullable Item c;",
+            "  @EnsuresNonNullIf(value=\"c\", result=true)",
+            "  // BUG: Diagnostic contains: postcondition inheritance is violated, this method must guarantee that all fields written in the @EnsuresNonNullIf annotation of overridden method SuperClass.hasA are @NonNull at exit point as well. Fields [a] must explicitly appear as parameters at this method @EnsuresNonNullIf annotation",
+            "  public boolean hasA() {",
+            "    return c != null;",
+            "  }",
+            "  @EnsuresNonNullIf(value={\"b\", \"c\"}, result=true)",
+            "  public boolean hasB() {",
+            "    return b != null && c != null;",
+            "  }",
+            "  public void doSomething() {",
+            "    if(hasB()) {",
+            "      b.call();",
+            "    }",
+            "  }",
+            "}")
+        .addSourceLines(
+            "Item.java", "package com.uber;", "class Item {", "  public void call() { }", "}")
+        .doTest();
+  }
+
+  @Test
+  public void setTrueIfNonNullToFalse() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Foo.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "import com.uber.nullaway.annotations.EnsuresNonNullIf;",
+            "class Foo {",
+            "  @Nullable Item nullableItem;",
+            "  @EnsuresNonNullIf(value=\"nullableItem\", result=false)",
+            "  public boolean doesNotHaveNullableItem() {",
+            "    return nullableItem == null;",
+            "  }",
+            "  public int runOk() {",
+            "    if(doesNotHaveNullableItem()) {",
+            "      return 1;",
+            "    }",
+            "    nullableItem.call();",
+            "    return 0;",
+            "  }",
+            "}")
+        .addSourceLines(
+            "Item.java", "package com.uber;", "class Item {", "  public void call() { }", "}")
+        .doTest();
+  }
+
+  @Test
+  public void setTrueIfNonNullToFalseMultipleElements() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Foo.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "import com.uber.nullaway.annotations.EnsuresNonNullIf;",
+            "class Foo {",
+            "  @Nullable Item nullableItem;",
+            "  @Nullable Item nullableItem2;",
+            "  @EnsuresNonNullIf(value={\"nullableItem\",\"nullableItem2\"}, result=false)",
+            "  public boolean doesNotHaveNullableItem() {",
+            "    return nullableItem == null && nullableItem2 == null;",
+            "  }",
```

Comment:
This annotation should _not_ pass validation.  If `nullableItem != null` but `nullableItem2 == null`, then this function returns `false`.  So returning `false` does not guarantee that both the fields are non-null.  I think using the `elseStore` (see comment above) will fix this problem.

Reply:
Your eyes are super trained for this!!! Great catch!